### DISCUSSION TITLE
How to contribute to urfave/cil/v2

### DISCUSSION THREAD

**Discussion by suzuki-shunsuke** [urfave/cli#1931] [Q&A] [answered]
Hello, thank you for your great project!

I want to contribute to this project, but I have a question about how to contribute.

I'm using urfave/cil/v2 and found an issue about shell completion, so I want to create an issue and send a pull request.
According to [the contribution guide](https://cli.urfave.org/CONTRIBUTING/), the main branch is compatible with urfave/cli/v2,
and changes incompatible with v2 should be added to `v3-dev-main`.
So I think I should send a pull request to the main branch.

But the main branch has a lot of changes from v2.27.2, and seems like incompatible changes are included.

https://github.com/urfave/cli/compare/v2.27.2...main

For example, I can't find the type `App` in the main branch.

```console
$ git rev-parse HEAD
c4cd0a51cc113a90eca318cc8f7d328f05a5f2b7

$ git grep "type App"
```

So I'm wondering the branch strategy of this project.

**Comment by suzuki-shunsuke** [comment]
> I'm using urfave/cil/v2 and found an issue about shell completion.

I created an issue.

- https://github.com/urfave/cli/issues/1932

**Comment by dearchap** [comment, marked as answer]
@suzuki-shunsuke The branches are 
1. v1-maint for v1
2. v2-maint for v2
3. main for v3 

v2 to v3 has breaking changes. v3 is much more feature rich than v2.

**Reply by suzuki-shunsuke** [reply]
Thank you.
